torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -877,13 +877,49 @@ Operator(                                                                      \
           };
         }),
     Operator(
-        "aten::__not__(int self) -> int",
+        "aten::__not__(int self) -> bool",
         [](Node* node) {
           return [=](Stack& stack) {
             push(stack, !pop(stack).toInt());
             return 0;
           };
         }),
+    Operator(
+        "aten::__not__(float self) -> bool",
```

Comment:
Need some discussion here, do we want "not" operator to be consistent with the python semantic as well? like `not []` or `not ""` is eval to True

Reply:
Can we just delete the overloads for anything that is not a float?

Reply:
Does it mean that we only use "not" for float? Since "not" is a boolean operator, should we only leave the bool overload instead and let every expr after "not" to use the "bool()" type cast?

Reply:
Sorry, I meant that we would delete everything but the boolean not operator as you suggest.